From fe8f20c27d34614ab668ccc540221f7c99ebcc84 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Thu, 20 Sep 2018 13:55:23 -0700
Subject: [PATCH 34/53] scsi: arcmsr: Remove unnecessary parentheses

Clang warns when multiple pairs of parentheses are used for a single
conditional statement.

drivers/scsi/arcmsr/arcmsr_hba.c:4138:19: warning: equality comparison
with extraneous parentheses [-Wparentheses-equality]
        if ((acb->dev_id == 0x1680)) {
             ~~~~~~~~~~~~^~~~~~~~~
drivers/scsi/arcmsr/arcmsr_hba.c:4138:19: note: remove extraneous
parentheses around the comparison to silence this warning
        if ((acb->dev_id == 0x1680)) {
            ~            ^        ~
drivers/scsi/arcmsr/arcmsr_hba.c:4138:19: note: use '=' to turn this
equality comparison into an assignment
        if ((acb->dev_id == 0x1680)) {
                         ^~
                         =
drivers/scsi/arcmsr/arcmsr_hba.c:4140:26: warning: equality comparison
with extraneous parentheses [-Wparentheses-equality]
        } else if ((acb->dev_id == 0x1880)) {
                    ~~~~~~~~~~~~^~~~~~~~~
drivers/scsi/arcmsr/arcmsr_hba.c:4140:26: note: remove extraneous
parentheses around the comparison to silence this warning
        } else if ((acb->dev_id == 0x1880)) {
                   ~            ^        ~
drivers/scsi/arcmsr/arcmsr_hba.c:4140:26: note: use '=' to turn this
equality comparison into an assignment
        } else if ((acb->dev_id == 0x1880)) {
                                ^~
                                =
drivers/scsi/arcmsr/arcmsr_hba.c:4164:26: warning: equality comparison
with extraneous parentheses [-Wparentheses-equality]
        } else if ((acb->dev_id == 0x1214)) {
                    ~~~~~~~~~~~~^~~~~~~~~
drivers/scsi/arcmsr/arcmsr_hba.c:4164:26: note: remove extraneous
parentheses around the comparison to silence this warning
        } else if ((acb->dev_id == 0x1214)) {
                   ~            ^        ~
drivers/scsi/arcmsr/arcmsr_hba.c:4164:26: note: use '=' to turn this
equality comparison into an assignment
        } else if ((acb->dev_id == 0x1214)) {
                                ^~
                                =
3 warnings generated.

Link: https://github.com/ClangBuiltLinux/linux/issues/146
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 drivers/scsi/arcmsr/arcmsr_hba.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/scsi/arcmsr/arcmsr_hba.c b/drivers/scsi/arcmsr/arcmsr_hba.c
index cc0be4651128..22cf697adab0 100644
--- a/drivers/scsi/arcmsr/arcmsr_hba.c
+++ b/drivers/scsi/arcmsr/arcmsr_hba.c
@@ -4135,9 +4135,9 @@ static void arcmsr_hardware_reset(struct AdapterControlBlock *acb)
 		pci_read_config_byte(acb->pdev, i, &value[i]);
 	}
 	/* hardware reset signal */
-	if ((acb->dev_id == 0x1680)) {
+	if (acb->dev_id == 0x1680) {
 		writel(ARCMSR_ARC1680_BUS_RESET, &pmuA->reserved1[0]);
-	} else if ((acb->dev_id == 0x1880)) {
+	} else if (acb->dev_id == 0x1880) {
 		do {
 			count++;
 			writel(0xF, &pmuC->write_sequence);
@@ -4161,7 +4161,7 @@ static void arcmsr_hardware_reset(struct AdapterControlBlock *acb)
 		} while (((readl(&pmuE->host_diagnostic_3xxx) &
 			ARCMSR_ARC1884_DiagWrite_ENABLE) == 0) && (count < 5));
 		writel(ARCMSR_ARC188X_RESET_ADAPTER, &pmuE->host_diagnostic_3xxx);
-	} else if ((acb->dev_id == 0x1214)) {
+	} else if (acb->dev_id == 0x1214) {
 		writel(0x20, pmuD->reset_request);
 	} else {
 		pci_write_config_byte(acb->pdev, 0x84, 0x20);
-- 
2.19.0

